### PR TITLE
Fix pulp_schedule and add tests

### PR DESCRIPTION
**Pull request by ekohl**
In 13385b7dbad852be906798ccdeb36a6df749cef6 I broke the pulp_schedule type. This fixes it, applies some more cleanups and adds tests to prevent future breakages.

Closes GH-260

### CODE REVIEW

**Comment by ccnifo** [comment]
I confirm 18a76e38eebcfd754c2b6bb92338d35ba7eb4bc1 fixes GH-260 on my setup, thanks.

**Comment by ekohl** [comment]
@ehelms would you mind having a look?

File: lib/puppet/provider/pulp_schedule/api.rb [hunk 1]
```diff
@@ -30,50 +30,42 @@ def initialize(resource={})
...
         $schedules_info['schedule_id'][schedule['_id']] = repo_id
       end
 
-      hash[:schedule_time] = schedule['schedule']
-      hash[:enabled] = schedule['enabled'] == true ? :true : :false
-      hash[:failure_threshold] = schedule['failure_threshold']
-      all << new(hash)
+      new({
+        :ensure => :present,
+        :provider => :pulp_schedule,
```
Comment (alexjfisher) [inline lib/puppet/provider/pulp_schedule/api.rb]: Is this right?  `pulp_schedule` is the type, but the provider is `api`?

Comment (ekohl) [inline lib/puppet/provider/pulp_schedule/api.rb]: I will admit that I only copied it from the old code and I'm a bit unsure of what it exactly does.

Comment (alexjfisher) [inline lib/puppet/provider/pulp_schedule/api.rb]: and I'll admit I'm only going by what I learnt from http://garylarizza.com/blog/2013/12/15/seriously-what-is-this-provider-doing/
I can't adequately explain why it works/doesn't seem to matter in this case.

Comment (ekohl) [inline lib/puppet/provider/pulp_schedule/api.rb]: I merged it now, but maybe we can drop it altogether. I don't really see it being used.

**Pull request merged**